.github/workflows/test-pull_request.yaml

```diff
@@ -156,3 +156,74 @@ jobs:
           else
               "./bin/$CONFIGURATION/unittests" --gtest_shuffle
           fi
+
+  build_mac:
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+        arch: [x86_64]
+    name: Mac
+    runs-on: macos-latest
+    steps:
+      - name: Cache Qt
+        id: cache-qt-lin
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/../Qt
+          key: ${{ runner.os }}-QtCache-${{ env.QT_VERSION }}
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          version: ${{ env.QT_VERSION }}
+          dir: ${{ github.workspace }}/..
+          cached: ${{ steps.cache-qt-lin.outputs.cache-hit }}
+          setup-python: 'false'
+          aqtversion: ==1.1.3
+
+      - uses: actions/checkout@v1
+        name: Checkout
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: ccache
```

Comment:
What is the purpose of ccache here?

Reply:
Same as with Linux builds – to speed up compilation by caching cache built object files.

Reply:
Ok, when should it be called?  I feel that if you're trying to cache object files here you should call the ccache action after they're built.

Reply:
`CMAKE_C_COMPILER_LAUNCHER` takes care of integrating ccache into the build, and the ccache Gitghub action has a pre- and post- action that downloads and uploads the object files respectively.

Reply:
(See the build log: it has a `Post ccache` step after the build.)